### PR TITLE
Exams by Sofiia Krytska IM-21

### PR DESCRIPTION
**Pull request by sonakrrr**
Refactored types.js and intersection.js

### CODE REVIEW

File: Tasks/types.js [hunk 1]
```diff
@@ -1,18 +1,17 @@
 // Count types in an array
+'use strict'
 
-types_ = function (s) {
-  types_ = {
+const types = (array) => {
+  const resObj = {
```
Comment (Vladyulik) [inline Tasks/types.js:5]: It would be better to rename `resObj` to just `result`

**Review by Vladyulik — CHANGES_REQUESTED** [review]

File: Tasks/intersection.js [hunk 2]
```diff
@@ -1,25 +1,15 @@
...
-      {
-      }
-    } else {
-      delete object_1[attribute_name];
-      {
-      }
-    }
-    {
+const intersection = (obj1, obj2) => {
+  const intersectionObj = {};
```
Comment (kovaliovev) [inline Tasks/intersection.js:5]: Consider renaming `intersectionObj` to just `result`

**Review by kovaliovev — CHANGES_REQUESTED** [review]

**Review by kovaliovev — APPROVED** [review]
All is good now!

**Review by Vladyulik — APPROVED** [review]
Now that's what I call a perfect code!

File: Tasks/intersection.js [hunk 3]
```diff
@@ -1,25 +1,15 @@
 // Find an intersection of two dictionaries
+'use strict'
```
Comment (tshemsedinov) [inline Tasks/intersection.js:2]:
```suggestion
'use strict';
```

File: Tasks/intersection.js [hunk 4]
```diff
@@ -1,25 +1,15 @@
...
-      {
-      }
-    }
-    {
+const intersection = (obj1, obj2) => {
+  const result = {};
+  const firstKeys = Object.keys(obj1);
+  for (const key of firstKeys) {
+    if (obj1[key] === obj2[key]) {
+      result[key] = obj1[key];
```
Comment (tshemsedinov) [inline Tasks/intersection.js:9]:
```suggestion
    const value = obj1[key];
    if (obj2[key] === value) {
      result[key] = value;
```

**Review by tshemsedinov — APPROVED** [review]
LGTM (looks good to me)

**Pull request closed without merging**